### PR TITLE
[E360-66] Package creation reusable workflow for AWS ECR

### PR DESCRIPTION
**Pull request by Dineshk77**
# Description:
- Require reusable workflow for publishing image to ECR

### CODE REVIEW

File: .github/workflows/package-creation-ecr.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+name: package-creation-ecr
+
+on:
+  workflow_call:
+    inputs:
+      aws_account_id:
+        description: "aws account number"
+        type: string
+        required: true
```
Comment (lawliet89) [inline .github/workflows/package-creation-ecr.yaml]: Let's combine this with `iam_role_name` to simply be the ARN of the IAM role.

**Review by lawliet89 — COMMENTED** [review]
As an improvement, maybe we can add in some Docker caching actions.

@niroz89 can you link @Dineshk77 up with your existing work on ECR image building perhaps?

File: .github/workflows/package-creation-ecr.yaml (hunk 1 above)
Comment (Dineshk77) [reply .github/workflows/package-creation-ecr.yaml]: Updated.

Comment (niroz89) [reply .github/workflows/package-creation-ecr.yaml]: can please take some reference here : https://github.com/SPHTech/bt-drupal9/blob/master/.github/workflows/deploy.yml

Comment (niroz89) [reply .github/workflows/package-creation-ecr.yaml]: And in prisma: 

https://github.com/SPHTech-Platform/reusable-workflows/blob/main/.github/workflows/prisma.yaml

Comment (niroz89) [reply .github/workflows/package-creation-ecr.yaml]: BT one both CI and CD are together, I'm gonna break that one, then will use the reusuable workflow.

File: .github/workflows/package-creation-ecr.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Extract branch name
+        uses: vazco/github-actions-branch-name@v1
+        id: branch
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        env:
+          REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          REPOSITORY: ${{ inputs.ecr_repository }}
+          IMAGE_TAG: ${{ github.sha }}
```
Comment (niroz89) [inline .github/workflows/package-creation-ecr.yaml]: sha may be too long.

Comment (Dineshk77) [reply .github/workflows/package-creation-ecr.yaml]: @niroz89 Any recommendation on differentiating the image tag per build?

Comment (niroz89) [reply .github/workflows/package-creation-ecr.yaml]: What I meant is shorter sha ha ha, We can just have this, and remove the line #57
```yaml
- name: Setup tag
  run: |
    echo "IMAGE_TAG=${GITHUB_SHA::8}" >> "${GITHUB_ENV}"
  shell: bash
```

Comment (niroz89) [reply .github/workflows/package-creation-ecr.yaml]: above will add into the github env, and availabe via `${{ env.IMAGE_TAG }}`

**Comment by niroz89** [comment]
> As an improvement, maybe we can add in some Docker caching actions.
> 
> @niroz89 can you link @Dineshk77 up with your existing work on ECR image building perhaps?

for buildx and caching, can take reference here.
https://github.com/SPHTech/bt-drupal9/blob/master/.github/workflows/deploy.yml
https://github.com/SPHTech-Platform/reusable-workflows/blob/main/.github/workflows/prisma.yaml

For BT, I have both cicd in same file now, will split it up and reuse it, have asked @pkailasu on this, he will add another reusable workflow for k8s-deploy.

https://github.com/SPHTech-Platform/docker-base-images/pull/1 needs update as well based on recent release ( https://github.com/aws-actions/amazon-ecr-login/pull/306 )

Eventually I will use the reusable workflow.

**Review by niroz89 — COMMENTED** [review]
@Dineshk77 bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.

**Comment by Dineshk77** [comment]
> bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.

Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?

**Comment by niroz89** [comment]
> > bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.
> 
> Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?

When we create ECR repo, it's recommended to specify the tags as immutable ( https://aquasecurity.github.io/tfsec/v0.63.1/checks/aws/ecr/enforce-immutable-repository/ ), means we won't be able to push with the same tag value, sha will make it dynamic, but when we hardcode `latest` tag, push will fail next time, if the repo set as immutable tag.

**Comment by Dineshk77** [comment]
> > bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.
> 
> Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?



> > bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.
> 
> Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?

I see I hardcoded the latest tag in the workflow. Will check on this

**Comment by Dineshk77** [comment]
> > > bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.
> > 
> > 
> > Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?
> 
> > > bro, one more thing, we may need to provide option to disable the latest tag, when we are creating ECR repo with immutable tags setting.
> > 
> > 
> > Hey @niroz89 , Can I get more info on this. Trying to understand the scenario. Is this more like deleting an image with tags?
> 
> I see I hardcoded the latest tag in the workflow. Will check on this

Added an input to set the tag mutability.

**Review by niroz89 — APPROVED** [review]

**Pull request merged**